### PR TITLE
[XEB] Split into three files

### PR DESCRIPTION
**Pull request by mpharrigan**
This PR is purely moving things around (on top of https://github.com/quantumlib/Cirq/pull/3760/) with one exception noted inline.

 - `random_quantum_circuit_generation` -- left alone; this is a well-scoped file
 - `xeb_sampling` - functionality related to sampling random circuits efficiently (for xeb)
 - `xeb_simulation` - functionality related to simulating 2-qubit circuits at various depths (for xeb) efficiently using multiprocessing
 - `xeb_fitting` - functionality related to using xeb to characterize gates by fitting parameters.

### CODE REVIEW

File: cirq/experiments/xeb_fitting.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    circuits: Sequence['cirq.Circuit'],
+    cycle_depths: Sequence[int],
+    param_resolver: 'cirq.ParamResolverOrSimilarType' = None,
+    pool: Optional['multiprocessing.pool.Pool'] = None,
+):
+    """Simulate and benchmark two-qubit XEB circuits.
+
+    This uses the estimator from
+    `cirq.experiments.fidelity_estimation.least_squares_xeb_fidelity_from_expectations`, but
+    adapted for use on pandas DataFrames for efficient vectorized operation.
```
Comment (mpharrigan) [inline cirq/experiments/xeb_fitting.py:55]: this is a new note

**Review by balopat — APPROVED** [review]

**Comment by balopat** [comment]
LGTM, previously reviewed on your fork. The reorg makes sense to me and the new note looks good too.

**Comment by mpharrigan** [comment]
I can't reproduce the notebook test failure locally. It looks unrelated, as it's in the "against stable cirq" tests, which should exclude all the xeb notebooks

**Comment by mpharrigan** [comment]
Might have been a race condition with merging #3781

**Comment by mpharrigan** [comment]
@mrwojtek I'm going to merge this so I can send out my next PR for you to review tonight (/your morning). If you have objections to this move, I can fix it up later!

**Pull request merged**